tidy3d/components/data/monitor_data.py

```diff
@@ -1392,13 +1413,21 @@ class ModeData(MonitorData):
         description="Complex-valued effective propagation constants associated with the mode.",
     )
 
-    n_group_raw: ModeIndexDataArray = pd.Field(
+    n_group_raw: GroupIndexDataArray = pd.Field(
```

Comment:
are there any potential side effects from changing this type? For example, in the json files, I think we store a tag with the DataArray type name, I wonder if we need any changes to the updater to convert these types if coming from older versions. On the other hand, if our IO tests pass, then it's probably working properly? just something that caught my attention as a possible issue.

Reply:
This should not be a problem. Older files have `n_group` as a `ModeIndexDataArray` but they can be loaded as `GroupIndexDataArray` because the only difference is in the description. I've just tested with a file saved in the base branch and it loaded correctly as a `GroupIndexDataArray` even though the hdf5 data has `ModeIndexDataArray` as the type from `n_group_raw`.

Reply:
sounds good